java-frontend/src/main/java/org/sonar/java/resolve/TypeSubstitution.java

```diff
@@ -28,6 +28,7 @@
 import java.util.Map;
 
 public class TypeSubstitution {
+  public boolean unchecked = false;
```

Comment:
Worth adding a few word to document what it means to have a unchecked "TypeSubstitution". Your example of `List` is good enough and would be good to document this change in code.